Added clone, pull & remove features

matiasberaldo opened the pull request:
This PR expands bos-workspace capabilities by introducing the following features:
- `bos-workspace clone <account_id>`: This command will download and create all the folder structure under `apps` for a given account
- `bos-workspace pull <account_id>`: When the cloning command is executed, the account gets added to a tree saved as `tree.json` in the root directory. Thanks to this, it is possible to know if your local version matches the one coming from the blockchain, and only write new or modified files.
- `bos-workspace remove <account_id>`: This command will delete the account components under the `app` directory. This method is necessary in order to refresh deleted accounts from the tree and keep it healthy.

Thanks!

matiasberaldo committed:
Added clone, pull & remove features

elliotBraem commented:
I've checked it out, it's a really convenient feature. I used it to initialize every.near; you can see that commit [here](https://github.com/NEARBuilders/every.near/commit/3718be08013e4c8e5682d89a3e5d2ba6f1029d19)

@sekaiking I'm wondering how this might be integrated with the v1.0.0? 

## Feedback
- [x] Clone should add a bos.config.json w/ appAccount so it can run in the workspace. See [this commit](25457c7ec45c2ea6067d8fc4d693debb21f566ec)
- [x] Is there a limit to number of files? I could clone devs.near fine (104 files), but get an error cloning big account like hack.near (300+)
<img width="567" alt="Screenshot 2024-01-27 at 8 01 33 PM" src="https://github.com/NEARBuilders/bos-workspace/assets/16282460/ee69f189-006e-4b07-8dc3-99be199f711c">

matiasberaldo committed:
Added support for downloading components from large accounts

matiasberaldo committed:
Create bos.config.json when cloning a project

matiasberaldo commented:
Hello @elliotBraem, thanks for your feedback! I have added the requested feature of initializing the `bos.config.json` file and added support for downloading a huge number of components from an account.

elliotBraem committed:
Merge branch 'main' into feature/download-and-sync-multiple-accounts

elliotBraem committed:
removes yarn.lock

elliotBraem committed:
removes yarn.lock

elliotBraem committed:
Update to node version 18

elliotBraem committed:
Remove --frozen-lockfile

elliotBraem committed:
convert to npm

elliotBraem committed:
lint and remove dist

elliotBraem committed:
lint ignore dist

elliotBraem committed:
remvoe dist

elliotBraem committed:
fix dist

elliotBraem committed:
fix natural language

elliotBraem committed:
clean gitignore
